spec.md

```diff
@@ -1616,23 +1617,34 @@ A register is initialized with an indeterminate value (see [@sec:indeterminate-v
 
 # Conditionals
 
-Conditional statements define one or more regions where the execution of operations (declarations, statements, and expressions) inside the regions is augmented by a *condition*.
-The evaluation behavior of operations is operation-dependent:
+Conditional statements define one or more regions where the operations (declarations, statements, and expressions) inside a region are *conditionally executed*.
+I.e., the operations within the region execute depending on the value of the condition and the kind of operation.
 
--   Connections (see [@sec:connections]) only drive their sink values when the product (logical and) of all conditions that are parents of the connect, but not parents of the sink declaration evaluates to true.
--   Commands (see [@sec:commands]) only evaluate when the product (logical and) of all parent conditions evaluates to true.
--   The evaluation of all other operations in conditional regions is unaffected by the condition.
-    E.g., register, wire, and node declarations as well as module instantiation statements and primitive operation expressions may occur in conditional regions but never evaluate conditionally.
+## Conditional Execution
```

Comment:
This section is probably good to have. I'm not sure that the initial version quite captures it.

If I have a module:

```firrtl
circuit Top :
  public module Top :
    output out : UInt<1>
    when UInt(1) :
      out := UInt(0) ; first connect
    else :
      out := UInt(1) ; second connect
```

Let's look at the first connect. The first connect is a conditional statement. There's only one parent of the connect (`UInt(1)`) and its product evaluates to true. The sink is `out` and it has no parents, so the product is vacuously true.

Combining these two, we see that the connect "only executes when `UInt(1)` is true, (which it always is), so this connect is always executing.

But what does that mean? 

* Are we assuming an event model, and the event constantly fires?
* Are we assuming an event model, and the event fires once, since it true at the start of the simulation but is a constant with "no sensitivity list"?
* Are we assuming this happens every clock cycle? (There's no clock, so this isn't it).

Reply:
This is a good question, though I'm not sure how much that this can be rectified on this PR as this seems to want a complete simulation / execution model of FIRRTL. I do think this is needed, though.

I think I'd try to define this as falling back to the definition of what connect means. The connect "is conditionally executed" (or evaluated, or simulated). What this conditional execution means should be left to the definition of the connect (which is not defined).

Taking a stab at this... what this means is that the connect changes the value of `out`. What "changes" mean and when it happens is a function of the destination. Wires are instantaneous and continuous. Registers are only updated on a clock. This pedantically _shouldn't_ be defined in terms of sensitivity lists, though it is is similar to how those work.